### PR TITLE
Flac 1.3.3

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **flac/1.3.3**

This is just a minor modification on top of @intelligide 's https://github.com/conan-io/conan-center-index/pull/577: the use of upstream CMakeLists.txt

Closes https://github.com/conan-io/conan-center-index/pull/577

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/flac/all/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(cmake_wrapper)
+
+if(EXISTS "${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+  include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+else()
+  include(conanbuildinfo.cmake)
+endif()
```
Comment (madebr) [inline recipes/flac/all/CMakeLists.txt]:
```suggestion
include(conanbuildinfo.cmake)
```

File: recipes/flac/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    _source_subfolder = "source_subfolder"
```
Comment (madebr) [inline recipes/flac/all/conanfile.py]:
```suggestion
    _cmake = None
    _source_subfolder = "source_subfolder"
```

File: recipes/flac/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            'set(CMAKE_EXE_LINKER_FLAGS -no-pie)',
+            '#set(CMAKE_EXE_LINKER_FLAGS -no-pie)')
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_EXAMPLES"] = False
+        cmake.definitions["BUILD_DOCS"] = False
+        cmake.definitions["BUILD_TESTING"] = False
+        cmake.configure()
+        return cmake
```
Comment (madebr) [inline recipes/flac/all/conanfile.py]: This avoids reconfiguration between `build` and `package`
```suggestion
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.definitions["BUILD_EXAMPLES"] = False
        self._cmake.definitions["BUILD_DOCS"] = False
        self._cmake.definitions["BUILD_TESTING"] = False
        self._cmake.configure()
        return self._cmake
```

File: recipes/flac/all/CMakeLists.txt (hunk 1 above)
Comment (Croydon) [reply recipes/flac/all/CMakeLists.txt]: Why this change? We added this to our templates because it seems necessary  for some workflows

**Comment by Croydon** [comment]
@ericLemanissier Could you please add `Closes https://github.com/conan-io/conan-center-index/pull/577` in the PR description? Thanks!

File: recipes/flac/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import os
+
+
+class FlacConan(ConanFile):
+    name = "flac"
+    description = "Free Lossless Audio Codec"
+    topics = ("conan", "flac", "codec", "audio", )
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/xiph/flac"
+    license = ("BSD", "GPL", "LPGL")
```
Comment (uilianries) [inline recipes/flac/all/conanfile.py]:
```suggestion
    license = ("BSD-3-Clause", "GPL-2.0-or-later", "LPGL-2.1-or-later", "GFDL-1.2")
```

File: recipes/flac/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        self.build_requires("nasm/2.14")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "{}-{}".format(self.name, self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, 'src', 'libFLAC', 'CMakeLists.txt'),
+            'target_link_libraries(FLAC PRIVATE $<$<BOOL:${HAVE_LROUND}>:m>)',
+            'target_link_libraries(FLAC PUBLIC $<$<BOOL:${HAVE_LROUND}>:m>)')
```
Comment (uilianries) [inline recipes/flac/all/conanfile.py]: Why do you need PUBLIC if you won't export that cmake file?

Comment (ericLemanissier) [reply recipes/flac/all/conanfile.py]: this package produces several libraries, which link to  the FLAC library. Without this change, they fail to link. The fix was merged upstream today: https://github.com/xiph/flac/pull/195

Comment (uilianries) [reply recipes/flac/all/conanfile.py]: Undertood. Thanks!

**Pull request merged**